libs/SalesforceSDKCore/SalesforceSDKCoreTests/WebSocketClientTests.swift

```diff
@@ -136,18 +136,22 @@ final class WebSocketClientTests: XCTestCase {
     }
     
     func testListenReceivesSuccessMessage() {
-        
         // Given
         mockTask = MockWebSocket()
         client = WebSocketClient(task: mockTask!)
         let expectation = self.expectation(description: "Message received")
-        
+        var fulfilled = false
         // When
         client?.listen { result in
             switch result {
             case .success(_):
                 // Then
-                expectation.fulfill()
+                if !fulfilled {
```

Comment:
I think we should investigate why it's being fulfilled more than once.

It looks like listen has a while loop to keep calling receive and the mock socket will always have a value, should it be cancelled after the first success?

Reply:
yeah, I'll look more into it.  Good call.